### DISCUSSION TITLE
Anchor git repositories' root in Path segment.

### DISCUSSION THREAD

**Discussion by ftpd** [JanDeDobbeleer/oh-my-posh#5186] [Show and tell]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What would you like to see changed?

Currently when I go into my git repository located somewhere on disk, the path looks like `~/git/projects/private/my-awesome-repository`. I would like to be able to see it as `my-awesome-repository`, because when I'm working inside the git repo, I don't really care about my global pwd, just the relation to repo's root.

Edit: I've just checked the default configuration and it seems it works that way, but I can't find related configuration option(s) in both `Path` and `Git` segment documentation. I'm sorry for creating an issue for this, I will close it as soon as I find the missing settings.

Edit2: ok, it seems that the default configuration uses style `Folder` and that's why I thought it covers my need - it doesn't. I want to see the full path everywhere except git repositories.

**Comment by boldandbrad** [comment]
I'm looking for something similar.

I'd like a variation of the `Folder` style called `GitFolder` or `GitRepo` that displays the current Folder but when inside a git repo, anchors the git repo root dir.

So if I'm in the root of my repo, lets say `path/to/dotfiles` it would should `dotfiles`, but if I cd into a sub folder within the repo, say `config/` then I'd like it to display `dotfiles/config` etc.

I can't find a way to force this behavior with omp today. Also happy to open a separate issue if what I'm asking for doesn't overlap with `ftpd`'s request.

**Comment by JanDeDobbeleer** [comment]
that's all possible with templates. I'll build one that does exactly that as I have the same behaviour.

**Comment by ftpd** [comment]
So, what's the next step? ;-)

**Comment by JanDeDobbeleer** [comment]
This is my template, which does what you're looking for:

```json
"template": "{{ $segment := .Segments.Git }}{{ $icon := \"\uf1d2\" }}{{ if $segment.Dir }} {{ $icon }} <i><b>{{ $segment.RepoName }}{{ if $segment.IsWorkTree}} \ue21c{{ end }}</b></i>{{if ne $segment.RepoName .Path}} \ueaf7{{ trimPrefix $segment.Dir .PWD }}{{ end }}{{ else }} \uea83 {{ path .Path .Location }}{{ end }}{{ if not .Writable }} \uf023{{ end }} ",
```

**Reply by boldandbrad** [reply]
Thanks for posting this! Was a huge help. I was able to convert it to toml format and modify it match what I like (no icons or writable indicator):

`template = ' at <b><p:blue>{{ $segment := .Segments.Git }}{{ if $segment.Dir }}{{ $segment.RepoName}}{{if ne $segment.RepoName .Path}}{{ trimPrefix $segment.Dir .PWD }}{{ end }}{{ else }}{{ path .Path .Location }}{{ end }}</></b> '`

**Comment by ftpd** [comment]
Thank you both - for a template and for rewriting it to toml (which I use).

I'm now all set, neat!
